### PR TITLE
Move common deployment helpers into common file

### PR DESCRIPTION
**Pull request by jsuchome**

### CODE REVIEW

File: script_library/deployment-actions-kvm.sh [hunk 1]
```diff
@@ -63,39 +36,6 @@ function remove_compute(){
...
     echo "Not implemented"
 }
-function clean_userfiles(){
-    echo "DANGER ZONE. Set the env var 'DELETE_ANYWAY' to delete everything in your userspace."
-    if [[ ${DELETE_ANYWAY:-"NO"} == "YES" ]]; then
-        extra_arg="-e delete_anyway='yes'"
-    fi
-    run_ansible ${socok8s_absolute_dir}/playbooks/generic-clean_userfiles.yml ${extra_arg:-}
-}
-function teardown(){
```
Comment (jhesketh) [inline script_library/deployment-actions-kvm.sh]: This is different to the openstack or common implementations.

File: script_library/deployment-actions-openstack.sh [hunk 2]
```diff
@@ -36,66 +37,6 @@ function enroll_caasp_workers() {
     echo "Enrolling caasp worker nodes into the cluster"
     run_ansible ${socok8s_absolute_dir}/playbooks/generic-enroll_caasp_workers.yml
 }
-function patch_upstream(){
-    echo "Running dev-patcher"
-    echo "Nothing will happen if developer mode is not set"
-    run_ansible ${socok8s_absolute_dir}/playbooks/openstack-patch_upstream.yml
```
Comment (jhesketh) [inline script_library/deployment-actions-openstack.sh]: This playbook (like some of the others following) just calls the generic playbook. We should remove the file if they are no longer needed.

File: script_library/deployment-actions-kvm.sh (hunk 1 above)
Comment (jsuchome) [reply script_library/deployment-actions-kvm.sh]: Yes, but I do not think adding `gather_logs` option into common part hurts.

File: script_library/deployment-actions-openstack.sh (hunk 2 above)
Comment (jsuchome) [reply script_library/deployment-actions-openstack.sh]: OK

**Comment by jsuchome** [comment]
@jhesketh updated

File: script_library/deployment-actions-kvm.sh (hunk 1 above)
Comment (jhesketh) [reply script_library/deployment-actions-kvm.sh]: No, but this is doing clean_kvm, and the new function does clean_openstack

File: script_library/deployment-actions-common.sh [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    fi
+    run_ansible ${socok8s_absolute_dir}/playbooks/generic-clean_userfiles.yml ${extra_arg:-}
+}
+
+function teardown(){
+    if [[ ${SOCOK8S_GATHER_LOGS:-"NO"} == "YES" ]]
+    then
+        gather_logs
+    fi
+    clean_openstack
```
Comment (jhesketh) [inline script_library/deployment-actions-common.sh]: This shouldn't be generic

File: script_library/deployment-actions-kvm.sh (hunk 1 above)
Comment (jsuchome) [reply script_library/deployment-actions-kvm.sh]: Ah, true, I missed that.

**Comment by Itxaka** [comment]
looks good but...should we wait to merge this until we cut the stable release in case we break something?

**Comment by toabctl** [comment]
retriggered CI.

**Comment by jsuchome** [comment]
@Itxaka or let's wait until the CI is green :-)

**Comment by jsuchome** [comment]
Fine. Better to use a tag then change request though ;-)

**Comment by jsuchome** [comment]
Rebased, ready for new reviews

**Pull request merged**